src/Adapter/AutoRouteRefreshAdapterInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2017 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\RoutingAutoBundle\Adapter;
+
+/**
+ * @author WAM Team <develop@wearemarketing.com>
+ */
+interface AutoRouteRefreshAdapterInterface
+{
+    /**
+     * Return all content of given class. Those classes will refresh their auto routes.
+     *
+     * @return object[] array
+     */
+    public function getAllContent(string $classFqn);
+
+    /**
+     * Return the identifier of an auto route.
+     *
+     * @param mixed $autoRouteableContent
+     *
+     * @return mixed
+     */
+    public function getIdentifier($autoRouteableContent);
```

Comment:
These are two methods could be resused indepently of one another, if we are segregating interfaces then these should be two seperate interfaces, otherwise we could just put them all in the `Adapter` interface. Not having worked on this for years, I would be tempted to just add them to the `AdapterInterface`?

We should return an [iterable](http://php.net/manual/en/language.types.iterable.php) type from `getAllContent` - it is very dangerous to perform f.e. an ODM query on 10,000 entities and hydrate them all as an array.